cmake/modules/hunter_windows_env_command.cmake

```diff
@@ -0,0 +1,15 @@
+# Copyright (c) 2015 Sumedh Ghaisas, Ruslan Baratov
+# All rights reserved.
+
+include(hunter_windows_to_snf_filename)
+
+#Set command to change environment variables to 8.3 filenames to windows
+function(hunter_windows_env_command variable_name)
+  
+  hunter_windows_to_snf_filename(${CMAKE_C_COMPILER} "CMAKE_C_COMPILER_SNF")
+  hunter_windows_to_snf_filename(${CMAKE_CXX_COMPILER} "CMAKE_CXX_COMPILER_SNF")
+  hunter_windows_to_snf_filename(${CMAKE_RC_COMPILER} "CMAKE_RC_COMPILER_SNF")
+  
+  set(${variable_name} "set" "cc=${CMAKE_C_COMPILER_SNF}" "&&" "set" "cxx=${CMAKE_CXX_COMPILER_SNF}" "&&" "set" "rc=${CMAKE_RC_COMPILER_SNF}" PARENT_SCOPE)
```

Comment:
Yes the code is needed because if you see the environment the build command is running in, it contains extra environment variables such as CC, CXX, RC etc. These variables are seemed to be set by external_project but I am not sure how or where. In case of Boost, it uses environment variable 'CC' which contains space and it fails. Secondly if we add more commands to BUILD_COMMAND by adding 'COMMAND' option, every command is run in new environment so setting environment that way won't help.